leshan-core/src/main/java/org/eclipse/leshan/core/request/DownLinkRequestVisitorAdapter.java

```diff
@@ -12,14 +12,15 @@
  * 
  * Contributors:
  *     Sierra Wireless - initial API and implementation
+ *     Michał Wadowski (Orange) - Add Observe-Composite feature.
  *******************************************************************************/
 package org.eclipse.leshan.core.request;
 
 /**
  * An abstract adapter class for visiting down link request. The methods in this class are empty. This class exists as
  * convenience for creating message observer objects.
  */
-public class DownLinkRequestVisitorAdapter implements DownlinkRequestVisitor {
+abstract public class DownLinkRequestVisitorAdapter implements DownlinkRequestVisitor {
```

Comment:
I don't see any instantiation of DownLinkRequestVisitorAdapter, and it's not used anywhere. What is the purpose of this class?

I changed it into abstract, because it seems like to be class that should not be instantiated in real code. But it's probably my not understanding the whole idea of this class, and in reality this change should not be even part of functionality I'm currently implementing :)

Reply:
> What is the purpose of this class?

The idea of this `Adapter` class is to propose an API which allow to implement the original Listener and only implement wanted methods. (I'm not sure I'm so clear :grimacing:)

So it could make sense to make this abstract  :thinking:  but as you said this is out of topic of this PR.
(Do not hesitate to do separated PR for this kinds of cleaning)